HesoyamHospital/GraphicEditor/Service/DrawingShapesService.cs

```diff
@@ -121,26 +117,29 @@ public Shape DrawShapes(GraphicalObject graphicalObject)
 
         public void DoubleClick(object sender, MouseButtonEventArgs e, string roomName)
         {
-            Rectangle rectangle = sender as System.Windows.Shapes.Rectangle;
-            List<InventoryItemDTO> result = new List<InventoryItemDTO>();
-            if (e.ClickCount == 2 && rectangle.Name.Contains("room"))
+            if (loggedIn.GetUserType() != UserType.PATIENT)
             {
-                RoomService roomService = Backend.AppResources.getInstance().roomService;
-                Room room = roomService.GetRoomByName(roomName);
-                long id = room.Id;
-
-                List<Medicine> medicine = (List<Medicine>)medicineService.GetMedicinesByRoom(id);
-                List<InventoryItem> inventoryItems = (List<InventoryItem>)inventoryService.GetInventoryItemsByRoom(roomName);
-
-                List<InventoryItemDTO> inventoryItemDTO = InvertoryItemMapper.ConvertFromIventoryItemToDTO(inventoryItems);
-                List<InventoryItemDTO> medicineDTO = InvertoryItemMapper.ConvertFromMedicineToDTO(medicine, roomName);
-                result.AddRange(inventoryItemDTO);
-                result.AddRange(medicineDTO);
-
-                OverviewEquipmentAndMedicine window = new OverviewEquipmentAndMedicine();
-                window.dataGridOverview.ItemsSource = result;
-                window.Show();
-
+                Rectangle rectangle = sender as System.Windows.Shapes.Rectangle;
+                List<InventoryItemDTO> result = new List<InventoryItemDTO>();
+                if (e.ClickCount == 2 && (rectangle.Name.Contains("room") || rectangle.Name.Contains("Storage")))
+                {
+                    RoomService roomService = Backend.AppResources.getInstance().roomService;
+                    Room room = roomService.GetRoomByName(roomName);
+                    long id = room.Id;
```

Comment:
Try breaking this section into one or two additional methods to increase readability.

Reply:
Thanks for the suggestion